src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheckTest.java

```diff
@@ -365,4 +365,20 @@ public void testReliefCommentBetweenMultipleComment() throws Exception {
                 getPath("InputFallThrough8.java"),
                 expected);
     }
+
+    @Test
```

Comment:
Please add another test case with similar inputs, except with switch labeled rules. See https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-SwitchRule

Reply:
> Please add another test case with similar inputs, except with switch labeled rules. See https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-SwitchRule

I am a bit confused. I am reading the documentation. And, it mentions of SwitchRule and SwitchLabel. 

```
int method(int a, int b) {
    top:
    switch (a) {
        case 1 -> {
            switch (b) {
                case 10 -> { break top; } // ok as terminates outer 'top' switch
                default -> { return 11; }
            }
        }
        case 2 -> { return 2; }
    }
}

```

Something like this? Like replacing with `->` ?

Reply:
> Like replacing with -> ?

Basically, yes. There are some different AST elements around this, so it will be good to make sure we are covered there.

Reply:
Just wanted to be clear here, there is no possibility of fallthrough with this syntax, I just want to make sure there is no possibility of NPE.

Reply:
> Basically, yes. There are some different AST elements around this, so it will be good to make sure we are covered there.

Have added a test-case and corresponding input file using switch rules. Let me know your thoughts on the same.

Reply:
Looks good. Thanks!